packager/media/formats/mp4/low_latency_segment_segmenter.cc

```diff
@@ -0,0 +1,235 @@
+  DCHECK(styp_);
+  
+  DCHECK(!sidx()->references.empty());
+  // earliest_presentation_time is the earliest presentation time of any access
+  // unit in the reference stream in the first subsegment.
+  sidx()->earliest_presentation_time =
+      sidx()->references[0].earliest_presentation_time;
+
+  if (options().segment_template.empty()) {
+    // Append the segment to output file if segment template is not specified.
+    file_name_ = options().output_file_name.c_str();
+  } else {
+    file_name_ = GetSegmentName(options().segment_template,
+                               sidx()->earliest_presentation_time,
+                               num_segments_, options().bandwidth);
+  }
+
+  // Create the segment file
+  segment_file_ = File::Open(file_name_.c_str(), "a");
+  if (!segment_file_) {
+    return Status(error::FILE_FAILURE, "Cannot open segment file: " +
+                                            file_name_);
+  }
+
+  std::unique_ptr<BufferWriter> buffer(new BufferWriter());
+  std::unique_ptr<File, FileCloser> file;
+
+  // Point to the already open segment file for writing
+  file.reset(segment_file_);
+  if (!file) {
+    return Status(error::FILE_FAILURE,
+                  "Cannot access the file to write " + file_name_);
+  }
+
+  // Write the styp header to the beginning of the segment.
+  styp_->Write(buffer.get());
+
+  const size_t segment_header_size = buffer->Size();
+  const size_t segment_size = segment_header_size + fragment_buffer()->Size();
+  DCHECK_NE(segment_size, 0u);
+
+  RETURN_IF_ERROR(buffer->WriteToFile(file.get()));
+  if (muxer_listener()) {
+    for (const KeyFrameInfo& key_frame_info : key_frame_infos()) {
+      muxer_listener()->OnKeyFrame(
+          key_frame_info.timestamp,
+          segment_header_size + key_frame_info.start_byte_offset,
+          key_frame_info.size);
+    }
+  }
+
+  // Write the chunk data to the file
+  RETURN_IF_ERROR(fragment_buffer()->WriteToFile(file.get()));
+
+  // Release the file to be used by the next chunk
+  file.release();
+
+  uint64_t segment_duration = GetSegmentDuration();
+  UpdateProgress(segment_duration);
+
+  if (muxer_listener()) {
+    if (!ll_dash_mpd_values_initialized_) {
+      // Set necessary values for LL-DASH mpd after the first chunk has been processed.
+      muxer_listener()->OnSampleDurationReady(sample_duration());
+      muxer_listener()->OnAvailabilityOffsetReady();
+      muxer_listener()->OnSegmentDurationReady();
+      ll_dash_mpd_values_initialized_ = true;
+    }
+    // Add the current segment in the manifest. 
+    // Following chunks will be appended to the open segment file.
+    muxer_listener()->OnNewSegment(file_name_,
+                                  sidx()->earliest_presentation_time,
+                                  segment_duration, segment_size);
+    is_initial_chunk_in_seg_ = false;
+  }
+
+  return Status::OK;
+}
+
+Status LowLatencySegmentSegmenter::WriteChunk() {
+  DCHECK(sidx());
```

Comment:
sidx isn't used in this method.

Reply:
Great catch! That was accidentally left over after I abstracted out the segment duration calculation to its own method `GetSegmentDuration()`.